### DISCUSSION TITLE
vpc to flow_log into an exiting s3 bucket

### DISCUSSION THREAD

**Discussion by asokolsky4clari** [gruntwork-io/knowledge-base#732] [How do I] [answered]
I am porting the code which relies on [AWS VPC module](https://github.com/terraform-aws-modules/terraform-aws-vpc) to rely on [gruntwork VPC module](https://github.com/gruntwork-io/terraform-aws-vpc/).  I would like for my vpc to flow_log into the existing s3 bucket.  In the former module this is a non-issue - I just provide an s3 arn.  [gruntwork vpc-flow-logs module](https://github.com/gruntwork-io/terraform-aws-vpc/blob/main/modules/vpc-flow-logs/main.tf) seems to be lacking such functionality.  

In other words, I need to:

* change [this line](https://github.com/gruntwork-io/terraform-aws-vpc/blob/main/modules/vpc-flow-logs/main.tf#L31) to allow for an externally provided s3 bucket arn
* provide a destination option [similar to this](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/vpc-flow-logs.tf#L28) to specify s3 flowlog file format - parquet.

Please advise.

---

<ins datetime="2023-06-07T15:52:21Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110239">Tracked in ticket #110239</a></p>
</ins>

**Comment by Etiene** [comment, marked as answer]
Hi @asokolsky4clari, there some options on how to go about this:

* forking our module and making any changes you find necessary
* forking, making the changes and openings a PR to us
* creating an issue in [the repo](https://github.com/gruntwork-io/terraform-aws-vpc) to request support for this feature

Question, are you currently blocked by the lack of this functionality?

**Comment by asokolsky4clari** [comment]
> creating an issue in [the repo](https://github.com/gruntwork-io/terraform-aws-vpc) to request support for this feature

done: https://github.com/gruntwork-io/terraform-aws-vpc/issues/313

> are you currently blocked by the lack of this functionality?

No.
As a workaround I created a wrapper module [as recommended](https://docs.gruntwork.io/guides/build-it-yourself/vpc/deployment-walkthrough/deploy-application-vp-cs), in which I create a vpc flow_log as I see fit.

Thanks

**Reply by Etiene** [reply]
Lovely thanks! This will go in our backlog then :)
